### PR TITLE
Abstract out the throttled iteration of requestDevices

### PR DESCRIPTION
**Pull request by Page-**
Change-type: patch

### CODE REVIEW

File: src/lib/device-proxy.ts [hunk 1]
```diff
@@ -136,6 +136,25 @@ export const proxy = async (req: Request, res: Response) => {
 	}
 };
 
+export async function throttledForEach<T, U>(
```
Comment (dfunckt) [inline src/lib/device-proxy.ts:139]: I'd rather this was exported from a `utils` module and received the delay to apply as an argument.

Comment (Page-) [reply src/lib/device-proxy.ts:139]: Urg, I did mean to make the delay configurable - I'll get that updated

Comment (Page-) [reply src/lib/device-proxy.ts]: done

File: src/lib/utils.ts [hunk 2]
```diff
@@ -56,3 +57,23 @@ export const varListInsert = (varList: EnvVarList, obj: Dictionary<string>) => {
...
+): Promise<U[]> => {
+	const promises: Array<PromiseLike<U>> = [];
+	for (const item of array) {
+		// We do not wait for each individual fn, we just throttle the calling of them
+		promises.push(fn(item));
+		// We add a delay between each notification so that we do not in essence
+		// trigger a DDOS from balena devices against us, but we do not wait for
+		// completion of individual requests because doing so could cause a
+		// terrible UX if we have a device time out, as that would block all the
+		// subsequent notifications
```
Comment (dfunckt) [inline src/lib/utils.ts:74]: This comment is better placed IMO at the call site in device-proxy.ts.

**Pull request merged**